src/imap-login/imap-login-client.c

```diff
@@ -389,6 +391,52 @@ static void imap_client_create(struct client *client, void **other_sets)
 				   IMAP_LOGIN_MAX_LINE_LENGTH);
 	if (imap_client->set->imap_literal_minus)
 		imap_parser_enable_literal_minus(imap_client->parser);
+
+	/* create our capability list from CAPABILITY_BANNER_STRING */
+	imap_client->capability_list =
+		 imap_capability_list_create(NULL);
+
+	if (*imap_client->set->imap_capability == '\0')
+		imap_capability_list_append_string(imap_client->capability_list,
+				CAPABILITY_BANNER_STRING);
+	else if (*imap_client->set->imap_capability != '+') {
+		imap_capability_list_append_string(imap_client->capability_list,
+				imap_client->set->imap_capability);
+		explicit_capability = TRUE;
+	} else {
+		/* add the capability banner string to the cap list */
+		imap_capability_list_append_string(imap_client->capability_list,
+				CAPABILITY_BANNER_STRING);
+		/* add everything after the plus to our cap list */
+		imap_capability_list_append_string(imap_client->capability_list,
+				imap_client->set->imap_capability + 1);
+	}
+
+	/* add STARTTLS to cap list if it is needed */
+	if (strcmp(client->ssl_set->ssl, "no") != 0) {
+		imap_capability_list_add(imap_client->capability_list,
+				"STARTTLS", IMAP_CAP_VISIBILITY_REQUIRE_ALL |
+							IMAP_CAP_VISIBILITY_PREAUTH |
+							IMAP_CAP_VISIBILITY_INSECURE);
+	}
+
+	/* add LOGINDISABLED to cap list */
+	imap_capability_list_add(imap_client->capability_list,
+			"LOGINDISABLED", IMAP_CAP_VISIBILITY_REQUIRE_ALL |
+							 IMAP_CAP_VISIBILITY_PREAUTH |
+							 IMAP_CAP_VISIBILITY_NO_LOGIN);
```

Comment:
Coding style consistency: Indentation here and elsewhere would be better as:
```
	imap_capability_list_add(imap_client->capability_list, "LOGINDISABLED",
		IMAP_CAP_VISIBILITY_REQUIRE_ALL |
		IMAP_CAP_VISIBILITY_PREAUTH |
		IMAP_CAP_VISIBILITY_NO_LOGIN);
```

(or if needed maybe put the strings to the second line with similar indentation)

Reply:
I may have misunderstood previous directions, are the parameters being passed to function calls supposed to line up with the "("? 

Or is that only for function definitions/declarations?

I only ask because the example code you give above doesn't seem to line up the capability flag parameters with the "(", but I'm not sure if that's just github rendering the tabs a certain way or not.

Reply:
Maybe the earlier comment I just sent made it clearer? It could line up to '(' as long as the line lengths don't go (much) over 80 chars. I'm not sure if they did here or not. Both are acceptable anyway :)